metis-indexing/src/main/java/eu/europeana/indexing/fullbean/RdfToFullBeanConverter.java

```diff
@@ -41,7 +41,7 @@ private static Date convertToDate(String dateString) {
     return Date.from(Instant.from(DateTimeFormatter.ISO_OFFSET_DATE_TIME.parse(dateString)));
   }
 
-  private static <S, T> List<T> convertList(List<S> sourceList, Function<S, T> converter,
+  protected static <S, T> List<T> convertList(List<S> sourceList, Function<S, T> converter,
```

Comment:
Changing the visibility of a method just for the sake of creating a test for it is considered bad practise. They are implementation details and public methods are calling those as well.

Reply:
Actually was not only for the sake of the test the intent was that the code needed to be testable, this indicates also that the abstraction is not located on the right place(file) a solution would be to abstract this methods to a separate class. After removing the unused code, there is no need to change the visibility so I am returning it to their previous state.

Reply:
It's arguable.
Generally public methods call private methods so by properly testing the public methods the private methods are tested as well. Unless those methods were not meant to be private or located in this class(e.g. also used by other classes)
Otherwise we would be making every method public so that we can test them individually which breaks it's encapsulation.

Reply:
Correct. this was just flagging an issue of design and you are right we don't want to break the encapsulation, and I agree that a proper testing will cover the private methods of a public method.